### PR TITLE
pi: initialize from PaliGemma with a scratch action expert

### PR DESCRIPTION
**Pull request by RyanPCo**
tests: follow transformers 5's is_offline_mode()

_hf_offline monkeypatched transformers.utils.hub._is_offline_mode, a module
flag that 5.x replaced with is_offline_mode(). setattr on a name the module no
longer has raises, so every test standing on the harness errored out before it
reached the network: 12 of them, including the whole PI train-step matrix.
Verified failing identically on an unmodified tree, so this is the venv's
transformers 5.17 meeting a 4.x assumption, not a regression from this stack.

Probe for whichever spelling the installed version has rather than pinning
either, since the lockfile still says 4.53.2.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01Xh3wMw9pXYPHUSGjU7qmix

pi: initialize from PaliGemma with a scratch action expert

An arm for training pi0.5 from the released VLM instead of the pi0.5 base
checkpoint, to separate what the VLM pretraining buys from what pi0.5's own
robot pretraining buys. The prefix (SigLIP tower, Gemma-2B, the projector)
comes from google/paligemma-3b-pt-224; everything pi0.5 adds on top -- the
Gemma-300M action expert, action_in/out_proj, the adaRMS time MLPs -- stays at
its init.

pytorch_weight_path and paligemma_weight_path are mutually exclusive rather
than ordered by precedence: silently preferring one would make a typo in the
other look like it worked. Setting neither now warns instead of loading
nothing quietly. The loader streams shard by shard and then fails if any
parameter went untouched or any checkpoint tensor went unused, because a
partial load of a 3.6B model is invisible in the loss for a long time.

Two places where the release does not line up with what openpi builds under
transformers 5, both handled in the key remap:

  - SiglipVisionModel was flattened, so vision_tower.vision_model.* is now
    vision_tower.*.
  - The checkpoint pads the vocabulary to 257216, 64 rows past the 257152
    openpi allocates (<image> plus 63 unused). embed_tokens and lm_head are
    truncated; openpi reaches the image slots through the vision tower, never
    through an embed_tokens lookup, so the dropped rows are unreachable.

google/paligemma-3b-pt-224 is license-gated, so resolve_paligemma_dir falls
back to ungated mirrors whose shards are sha256-checked against the canonical
repo's LFS metadata first -- a mirror that has drifted is refused, not used.
paligemma_allow_mirror: false turns the fallback off. openpi's own loader
source is not an option: its GCS bucket 403s anonymously.

pi0.5_paligemma_init is an overlay, not a runnable config, so a keypoint or 6D
arm is four lines. It changes the init source only; the constant LR stays,
since pi0.5_base documents dropping the cosine schedule on purpose.

openpi_compat carries two shims for transformers 5, which this venv has and
openpi's pin (4.53.2) does not expect. They are load-bearing on the pi05_base
path too -- without them pi does not run at all here -- and both should be
deleted once the fork's transformers_replace patch is ported forward. Left
alone, and documented: openpi's fp32 keep-list still spells the pre-flattening
vision paths, so the patch and position embeddings run in bf16 on both init
paths. Pre-existing, not this arm's to change.

Verified with a full trainHydra run to exit 0 on an H100 (4 epochs, all four
validation heads, checkpoint and videos written), the real 12 GB checkpoint
into the real model in tests/integration, and 42 unit tests.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01Xh3wMw9pXYPHUSGjU7qmix

### CODE REVIEW

**Comment by RyanPCo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/677?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#680** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/680?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#681](https://github.com/GaTech-RL2/EgoVerse/pull/681) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/681?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#682](https://github.com/GaTech-RL2/EgoVerse/pull/682) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/682?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#679** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/679?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#678** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#677** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/677?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/677?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#676** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/676?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#674** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/674?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#673** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/673?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#672** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/672?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#671** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/671?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#670** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/670?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#669** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/669?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#668** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/668?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#667** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#666** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/666?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#665** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/665?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#664** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/664?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#646** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/646?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#647](https://github.com/GaTech-RL2/EgoVerse/pull/647) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/647?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#645** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/645?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#650** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/650?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#649** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/649?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#652** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/652?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#651** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/651?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#644** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/644?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#643** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/643?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#642** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/642?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#640** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/640?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#659** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/659?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#658** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/658?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#657** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/657?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#656** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/656?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#655** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/655?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#654** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/654?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#653** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/653?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#638** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/638?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#639](https://github.com/GaTech-RL2/EgoVerse/pull/639) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/639?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#637** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/637?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#636** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/636?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#634** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/634?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#635](https://github.com/GaTech-RL2/EgoVerse/pull/635) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/635?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#633** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/633?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#631** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/631?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#632](https://github.com/GaTech-RL2/EgoVerse/pull/632) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/632?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#630** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/630?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#628** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/628?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#629** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/629?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#626** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/626?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#625** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/625?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#624** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/624?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#623** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/623?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#622** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/622?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#627** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/627?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#621](https://github.com/GaTech-RL2/EgoVerse/pull/621) <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/621?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#620** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/620?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#619** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
